snapcraft/_store.py

```diff
@@ -404,6 +425,9 @@ def push_metadata(snap_filename, force):
     with _requires_login():
         store.push_precheck(snap_name)
         store.push_metadata(snap_name, metadata, force)
+        with _get_icon_from_snap_file(snap_filename) as icon:
+            metadata = {'icon': icon}
+            store.push_binary_metadata(snap_name, metadata, force)
```

Comment:
It's possible for the `icon` to be `None`, here. I can't tell from the below: will that remove the icon from the store? That situation does not seem covered by tests.

Reply:
Yes, `icon` could be None, and that will remove the icon from the store.
I'll review tests and add it if missing. Thanks!